cristin-commons/src/main/java/no/unit/nva/cristin/common/client/IQueryApiClient.java

```diff
@@ -0,0 +1,10 @@
+package no.unit.nva.cristin.common.client;
+
+import no.unit.nva.cristin.model.SearchResponse;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+
+public interface IQueryApiClient<T, R> {
+
+    SearchResponse<R> executeQuery(T params) throws ApiGatewayException;
```

Comment:
Dersom T alltid vil være `Map<String, String>` slik det ser ut for meg, trenger den ikke å være generic.

Reply:
Og dersom vi ikke har javadoc, bruk gode navn på generics, feks:
`IQueryApiClient<T, DTO>`

Reply:
Ved utvidelse av bruken av denne kan vi feks måtte sende inn et objekt av type QueryProject, så jeg tenker det kan være lurt å ha den generisk

Reply:
https://pmd.sourceforge.io/pmd-6.20.0/pmd_rules_java_codestyle.html#genericsnaming